File: api/src/main/java/com/wheats/api/mypage/dto/SupportTicketResponse.java
package com.wheats.api.mypage.dto;

import com.wheats.api.mypage.entity.SupportTicketEntity;
import com.wheats.api.mypage.entity.SupportTicketStatus;

import java.time.LocalDateTime;

public class SupportTicketResponse {

    private Long id;
    private Long storeId;
    private String storeName;
    private String title;
    private String message;
    private SupportTicketStatus status;
    private LocalDateTime createdAt;

    public SupportTicketResponse(Long id,
                                 Long storeId,
                                 String storeName,
                                 String title,
                                 String message,
                                 SupportTicketStatus status,
                                 LocalDateTime createdAt) {
        this.id = id;
        this.storeId = storeId;
        this.storeName = storeName;
        this.title = title;
        this.message = message;
        this.status = status;
        this.createdAt = createdAt;
    }

    public static SupportTicketResponse from(SupportTicketEntity entity, String storeName) {
        return new SupportTicketResponse(
                entity.getId(),
                entity.getStoreId(),
                storeName,
                entity.getTitle(),
                entity.getMessage(),
                entity.getStatus(),
                entity.getCreatedAt()
        );
    }

    public Long getId() {
        return id;
    }

    public Long getStoreId() {
        return storeId;
    }

    public String getStoreName() {
        return storeName;
    }

    public String getTitle() {
        return title;
    }

    public String getMessage() {
        return message;
    }

    public SupportTicketStatus getStatus() {
        return status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
}

File: mobile/app/src/main/java/com/example/mobile/data/model/SupportTicketResponse.kt
package com.example.mobile.data.model

/**
 * 1:1 문의 응답 DTO
 * - 서버의 SupportTicketResponse(Java)와 필드 구조를 맞춘 버전
 */
data class SupportTicketResponse(
    val id: Long,
    val storeId: Long,
    val title: String,
    val message: String,
    val status: String,   // 서버에서는 enum(OPEN/ANSWERED/CLOSED)을 문자열로 내려온다고 가정
    val createdAt: String // ISO 8601 날짜 문자열
)


